src/Components/test/E2ETest/Tests/InteropTest.cs

```diff
@@ -63,6 +63,7 @@ public void CanInvokeDotNetMethods()
                 ["roundTripByteArrayWrapperObjectAsyncFromJS"] = @"{""strVal"":""Some string"",""byteArrayVal"":{""0"":1,""1"":5,""2"":7,""3"":17,""4"":200,""5"":138},""intVal"":42}",
                 ["roundTripByteArrayAsyncFromDotNet"] = @"1,5,7,15,35,200",
                 ["roundTripByteArrayWrapperObjectAsyncFromDotNet"] = @"StrVal: Some String, IntVal: 100000, ByteArrayVal: 1,5,7,15,35,200",
+                ["jsToDotNetStreamAsync"] = "Success",
```

Comment:
Are there also E2E test cases covering errors? I know it's quite hard to simulate them. Maybe there aren't even any cases where the transfer can reasonably fail other than if the SignalR connection drops.

To what extent can we check that the .NET side reading the stream gets an exception if the SignalR connection drops midway?

Reply:
Actually, *would* there be an error in that case? Would the `ReadAsync` call on the .NET side eventually time out, or does it hang forever?

Reply:
We can simulate some of these cases with Ignitor. 

We also need to account for cases where the client stops sending data after we blocked to read, in which case we should probably terminate the circuit?

Reply:
We're trying to deprecate and remove Ignitor, aren't we? (cc @captainsafia)

If it's possible to simulate through either unit or E2E tests that would be ideal, as we might not have Ignitor for much longer.

Reply:
> We're trying to deprecate and remove Ignitor, aren't we? (cc @captainsafia)

Yep -- the plan is to move to unit tests for the ComponentHub where possible. We have [some tests that validate timeouts during JS interop on the server](https://github.com/dotnet/aspnetcore/blob/main/src/Components/test/E2ETest/ServerExecutionTests/ServerInteropTestJsInvocationsTimeoutsBehavior.cs) but not sure if they will cover the scenario here.

Reply:
> We also need to account for cases where the client stops sending data after we blocked to read, in which case we should probably terminate the circuit?

How about a 10 second timeout between chunks. If we don't receive the next chunk by the timeout, we throw an error/dispose the stream?

Reply:
Added 1 minute timeout, after which if we haven't received any new chunks we terminate the stream.
https://github.com/dotnet/aspnetcore/pull/33491/commits/381c53d2e2336cb6de71f22314f7e9f5af2511e4

Also added chunk sequence validation to guard against SignalR Disconnects.
https://github.com/dotnet/aspnetcore/pull/33491/commits/6e09532bf49eed0f404f3d95e83f49f28849b2cb